tests/journal_lock.py/test_journal_lock.py

```diff
@@ -0,0 +1,349 @@
+from py._path.local import LocalPath as py_path_local_LocalPath
+
+from config import config
+from journal_lock import JournalLock, JournalLockResult
+
+
+# We need another process to already hold the lock.
+def other_process_lock(continue_q: mp.Queue, exit_q: mp.Queue, lockfile: pathlib.Path):
+    """
+    Obtain the lock in a sub-process.
+
+    :param continue_q: Write to this when parent should continue.
+    :param exit_q: When there's an item in this, exit.
+    :param lockfile: Path where the lockfile should be.
+    """
+    with open(lockfile / 'edmc-journal-lock.txt', mode='w+') as lf:
+        print(f'sub-process: Opened {lockfile} for read...')
+        # This needs to be kept in sync with journal_lock.py:_obtain_lock()
+        if not _obtain_lock('sub-process', lf):
+            print('sub-process: Failed to get lock, so returning')
+            return
+
+        print('sub-process: Got lock, telling main process to go...')
+        continue_q.put('go', timeout=5)
+        # Wait for signal to exit
+        print('sub-process: Waiting for exit signal...')
+        exit_q.get(block=True, timeout=None)
+
+
+def _obtain_lock(prefix: str, filehandle) -> bool:
+    """
+    Obtain the JournalLock.
+
+    :param prefix: str - what to prefix output with.
+    :param filehandle: File handle already open on the lockfile.
+    :return: bool - True if we obtained the lock.
+    """
+    if sys.platform == 'win32':
+        print(f'{prefix}: On win32')
+        import msvcrt
+        try:
+            print(f'{prefix}: Trying msvcrt.locking() ...')
+            msvcrt.locking(filehandle.fileno(), msvcrt.LK_NBLCK, 4096)
+
+        except Exception as e:
+            print(f'{prefix}: Unable to lock file: {e!r}')
+            return False
+
+    else:
+        import fcntl
+
+        print(f'{prefix}: Not win32, using fcntl')
+        try:
+            fcntl.flock(filehandle, fcntl.LOCK_EX | fcntl.LOCK_NB)
+
+        except Exception as e:
+            print(f'{prefix}: Unable to lock file: {e!r}')
+            return False
+
+    return True
+
+
+class TestJournalLock:
+    """JournalLock test class."""
+
+    @pytest.fixture
+    def mock_journaldir(self, monkeypatch: _pytest_monkeypatch, tmpdir: _pytest_tmpdir) -> py_path_local_LocalPath:
+        """Fixture for mocking config.get_str('journaldir')."""
+        def get_str(key: str, *, default: str = None) -> str:
+            """Mock config.*Config get_str to provide fake journaldir."""
+            if key == 'journaldir':
+                return tmpdir
+
+            print('Other key, calling up ...')
+            return config.get_str(key)  # Call the non-mocked
+
+        with monkeypatch.context() as m:
+            m.setattr(config, "get_str", get_str)
+            yield tmpdir
+
+    @pytest.fixture
+    def mock_journaldir_changing(
+            self,
+            monkeypatch: _pytest_monkeypatch,
+            tmpdir_factory: _pytest_tmpdir.TempdirFactory) -> py_path_local_LocalPath:
```

Comment:
```suggestion
            tmpdir_factory: _pytest_tmpdir.TempdirFactory
        ) -> py_path_local_LocalPath:
```
Not 100% on this one, I just didnt at a glance see where the function declaration ended and the body begun

Reply:
You're right, once we start line-separating function definition args like that we should separate the return type as well.